### PR TITLE
Feature/remote components default values

### PR DESCRIPTION
**Pull request by jakubneubauer**
Created tranformer instance can initialize its properties in constructor. These initial values are then visible by user and he regards them as the default values. We want similar behaviour when using the remote components.
So server must provide these values in component info and client should use them when creating new instance of remote transformer.

### CODE REVIEW

File: engine/remote-components/src/main/java/org/datacleaner/descriptors/RemoteDescriptorProvider.java [hunk 1]
```diff
@@ -102,25 +110,27 @@ private void downloadDescriptors() {
...
                                         cl,
                                         propInfo.isRequired(),
                                         transformer,
-                                        initAnnotations(component.getName(), propertyName, propInfo.getAnnotations())));
+                                        initAnnotations(component.getName(), propertyName, propInfo.getAnnotations()), propInfo.getDefaultValue()));
                             } catch(Exception e) {
+                                logger.debug("Cannot initialize typed property descriptor '{}'.'{}'", component.getName(), propertyName, e);
                                 // class not available on this server.
-                                transformer.addPropertyDescriptor(new JsonSchemaConfiguredPropertyDescriptorImpl(
+                                transformer.addPropertyDescriptor(propDesc = new JsonSchemaConfiguredPropertyDescriptorImpl(
```
Comment (LosD) [inline engine/remote-components/src/main/java/org/datacleaner/descriptors/RemoteDescriptorProvider.java]: Any reason for adding propDesc? It's not used for anything (besides feeding directly into addPropertyDescriptor, of course).

File: engine/remote-components/src/main/java/org/datacleaner/descriptors/RemoteDescriptorProvider.java [hunk 2]
```diff
@@ -151,4 +165,7 @@ private void downloadDescriptors() {
         return annotations;
     }
 
+    public static void main(String[] args) throws Exception {
+    }
+
```
Comment (LosD) [inline engine/remote-components/src/main/java/org/datacleaner/descriptors/RemoteDescriptorProvider.java]: Errrr? :)

**Comment by LosD** [comment]
Looks good to me, but see comments

**Comment by jakubneubauer** [comment]
The comments were incorporated.

**Comment by LosD** [comment]
Super, let's merge when Travis is happy :)

**Comment by LosD** [comment]
And Travis is happy now. Merging!

**Pull request merged**